app/src/main/java/org/jellyfin/mobile/utils/PermissionRequestHelper.kt

```diff
@@ -0,0 +1,50 @@
+package org.jellyfin.mobile.utils
+
+import android.app.Activity
+import android.content.pm.PackageManager
+import androidx.core.app.ActivityCompat
+import androidx.core.app.ActivityCompat.requestPermissions
+import org.koin.android.ext.android.getKoin
+
+class PermissionRequestHelper {
+    private val permissionRequests = mutableMapOf<Int, (Map<String, Int>) -> Unit>()
```

Comment:
I just remembered, for the callback you could introduce a typealias if you want. Would make the code even neater :D